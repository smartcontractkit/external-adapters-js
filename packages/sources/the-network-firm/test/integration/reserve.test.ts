import {
  setEnvVariables,
  TestAdapter,
} from '@chainlink/external-adapter-framework/util/testing-utils'
import * as nock from 'nock'
import {
  mockM0ResponseSuccess,
  mockReResponseSuccess,
  mockReserveEmgemxResponseSuccess,
  mockReserveUraniumResponseSuccess,
} from './fixtures'

describe('execute', () => {
  let spy: jest.SpyInstance
  let testAdapter: TestAdapter
  let oldEnv: NodeJS.ProcessEnv

  beforeAll(async () => {
    oldEnv = JSON.parse(JSON.stringify(process.env))
    const mockDate = new Date('2001-01-01T11:11:11.111Z')
    spy = jest.spyOn(Date, 'now').mockReturnValue(mockDate.getTime())

    process.env.ALT_API_ENDPOINT = 'http://test-endpoint-new'
    process.env.URANIUM_DIGITAL_QOHMMJQAF4JK_API_KEY = 'uranium-api-key'
    process.env.EMGEMX_TDFKF3_API_KEY = 'emgemx-api-key'
    process.env.M0_STABLECOIN_INPD83_API_KEY = 'm0-api-key'
    process.env.RE_PROTOCOL_8TAWLM_API_KEY = 're-api-key'

    const adapter = (await import('../../src')).adapter
    adapter.rateLimiting = undefined
    testAdapter = await TestAdapter.startWithMockedCache(adapter, {
      testAdapter: {} as TestAdapter<never>,
    })
  })

  afterAll(async () => {
    setEnvVariables(oldEnv)
    await testAdapter.api.close()
    nock.restore()
    nock.cleanAll()
    spy.mockRestore()
  })

  describe('reserve endpoint', () => {
    it('should return success for uranium', async () => {
      const data = {
        endpoint: 'reserve',
        client: 'uranium-digital-qohmmjqaf4jk',
      }

      mockReserveUraniumResponseSuccess()

      const response = await testAdapter.request(data)
      expect(response.statusCode).toBe(200)
      expect(response.json()).toMatchSnapshot()
    })

    it('should return success for emgemx', async () => {
      const data = {
        endpoint: 'reserve',
        client: 'emgemx-tdfkf3',
      }

      mockReserveEmgemxResponseSuccess()

      const response = await testAdapter.request(data)
      expect(response.statusCode).toBe(200)
      expect(response.json()).toMatchSnapshot()
    })

    it('should return success for m0', async () => {
      const data = {
        endpoint: 'reserve',
        client: 'm0-stablecoin-inpd83',
      }

      mockM0ResponseSuccess()

      const response = await testAdapter.request(data)
      expect(response.statusCode).toBe(200)
      expect(response.json()).toMatchSnapshot()
    })

    it('should return success for re', async () => {
      const data = {
        endpoint: 'reserve',
        client: 're-protocol-8tawlm',
      }

      mockReResponseSuccess()

      const response = await testAdapter.request(data)
      expect(response.statusCode).toBe(200)
      expect(response.json()).toMatchSnapshot()
    })

    it('should fail if client name is not present or wrong', async () => {
      const data = {
        endpoint: 'reserve',
      }

      const response = await testAdapter.request(data)
      expect(response.statusCode).toBe(400)
      expect(response.json()).toMatchSnapshot()
    })
  })
})
