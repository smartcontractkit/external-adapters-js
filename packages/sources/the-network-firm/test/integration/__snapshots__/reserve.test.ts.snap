// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute reserve endpoint should fail if client name is not present or wrong 1`] = `
{
  "error": {
    "message": "[Param: client] param is required but no value was provided",
    "name": "AdapterError",
  },
  "status": "errored",
  "statusCode": 400,
}
`;

exports[`execute reserve endpoint should return success for emgemx 1`] = `
{
  "data": {
    "result": "1000000",
    "ripcord": false,
  },
  "result": "1000000",
  "statusCode": 200,
  "timestamps": {
    "providerDataReceivedUnixMs": 978347471111,
    "providerDataRequestedUnixMs": 978347471111,
    "providerIndicatedTimeUnixMs": 1751500917131,
  },
}
`;

exports[`execute reserve endpoint should return success for m0 1`] = `
{
  "data": {
    "result": "1000000",
    "ripcord": false,
  },
  "result": "1000000",
  "statusCode": 200,
  "timestamps": {
    "providerDataReceivedUnixMs": 978347471111,
    "providerDataRequestedUnixMs": 978347471111,
    "providerIndicatedTimeUnixMs": 1751500917131,
  },
}
`;

exports[`execute reserve endpoint should return success for re 1`] = `
{
  "data": {
    "result": "1000000",
    "ripcord": false,
  },
  "result": "1000000",
  "statusCode": 200,
  "timestamps": {
    "providerDataReceivedUnixMs": 978347471111,
    "providerDataRequestedUnixMs": 978347471111,
    "providerIndicatedTimeUnixMs": 1751500917131,
  },
}
`;

exports[`execute reserve endpoint should return success for uranium 1`] = `
{
  "data": {
    "result": "1000000",
    "ripcord": false,
  },
  "result": "1000000",
  "statusCode": 200,
  "timestamps": {
    "providerDataReceivedUnixMs": 978347471111,
    "providerDataRequestedUnixMs": 978347471111,
    "providerIndicatedTimeUnixMs": 1751500917131,
  },
}
`;